--- evince-2.28.2/previewer/ev-previewer.c	2009-12-14 20:47:36.000000000 +0100
+++ evince-2.28.2/previewer/ev-previewer.c	2010-05-12 16:14:42.000000000 +0200
@@ -85,8 +85,12 @@ ev_previewer_load_document (const gchar 
 {
 	EvJob *job;
 	gchar *uri;
+	GFile  *file;
+
+	file = g_file_new_for_commandline_arg (filename);
+	uri = g_file_get_uri (file);
+	g_object_unref (file);
 
-	uri = g_filename_to_uri (filename, NULL, NULL);
 	job = ev_job_load_new (uri);
 	g_signal_connect (job, "finished",
 			  G_CALLBACK (ev_previewer_load_job_finished),
